### PR TITLE
fixes the typedIdent problem for good

### PR DESCRIPTION
**Pull request by xeno-by**
Previous attachment retaining fix was only working for Idents which
get turned into Selects. Now it works for all transformations applied
to Idents (e.g. when an ident refers to something within a package obj).

### CODE REVIEW

**Comment by xeno-by** [comment]
A follow-up pull request for https://github.com/scala/scala/pull/1766, which unfortunately failed to completely fix the problem.

**Comment by xeno-by** [comment]
review @paulp @retronym

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -4767,7 +4767,7 @@ trait Typers extends Modes with Adaptations with Tags {
               // SI-5967 Important to replace param type A* with Seq[A] when seen from from a reference, to avoid
               //         inference errors in pattern matching.
               stabilize(tree2, pre2, mode, pt) modifyType dropIllegalStarTypes
-            }
+            }) setAttachments tree.attachments
         }
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala:4771]: I think this one isn't done until you only have to call "setAttachments tree.attachments" from one place. We should especially resist having multiple calls to it when one is nested inside another one...

Comment (xeno-by) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala:4771]: Let's please assume this didn't happen

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by xeno-by** [comment]
Does this look good now?

**Comment by paulp** [comment]
Yes, lgtm.

**Pull request merged**